mono/metadata/threads.c

```diff
@@ -4485,24 +4524,36 @@ static void CALLBACK dummy_apc (ULONG_PTR param)
 #endif
 
 /*
- * mono_thread_execute_interruption
+ * mono_thread_execute_interruption_handle
  * 
  * Performs the operation that the requested thread state requires (abort,
  * suspend or stop)
  */
-static MonoException*
-mono_thread_execute_interruption (void)
+static MonoExceptionHandle
+mono_thread_execute_interruption_handle (void)
 {
-	MonoInternalThread *thread = mono_thread_internal_current ();
-	MonoThread *sys_thread = mono_thread_current ();
+// FIXME This function does unnecessary allocations and handle indirections.
+// Not every path needs every handle allocated.
+// Internal thread is pinned. The integer field uses do not need handles.
 
-	LOCK_THREAD (thread);
+	HANDLE_FUNCTION_ENTER ();
+
+	MonoInternalThreadHandle thread = mono_thread_internal_current_handle ();
+	MonoThreadObjectHandle sys_thread = mono_thread_current_handle ();
+	MONO_HANDLE_LOCAL_VARIABLE_INITIALIZED_NULL (MonoException, exc);
```

Comment:
This is hardly readable. What does it bring over the existing syntax of `TYPED_HANDLE_NAME(type) name = MONO_HANDLE_NEW (type, NULL)`?

Reply:
A little more exact and less repetitive -- don't need to repeat type, for one thing.

Reply:
It is very readable because it is so exact, almost a sentence or comment -- local variable initialized (to) NULL. It does hide "new" though.

Reply:
Can you find this pattern in use in the rest of the code base? Is it a common pattern in to magically initialize a local variable when declaring it in C (remember, this is not C++)?

Reply:
Yes it is common pattern in C as well. We should consider adopting new idioms sometimes.

Reply:
Oh you meant the macro, not the { 0 }. In that case, well, yes, macros to capture recurring patterns is what macros are for. And also to name idioms with something that might be shorter or quicker to read. It is a bit subjective, because it also helps to have a smaller common vocabularly and repeatedly build from it, than to constantly invent new terms for new scenarios (like error_init).